swarm/storage/dbstore.go

```diff
@@ -95,10 +95,19 @@ func NewDbStore(path string, hash SwarmHasher, capacity uint64, radius int) (s *
 
 	data, _ := s.db.Get(keyEntryCnt)
 	s.entryCnt = BytesToU64(data)
+	if len(data) > 0 {
+		s.entryCnt++
+	}
 	data, _ = s.db.Get(keyAccessCnt)
 	s.accessCnt = BytesToU64(data)
+	if len(data) > 0 {
+		s.accessCnt++
+	}
 	data, _ = s.db.Get(keyDataIdx)
 	s.dataIdx = BytesToU64(data)
+	if len(data) > 0 {
+		s.dataIdx++
+	}
```

Comment:
Why is an increase in these indexes needed after (as I understand) just reading its very value?

Reply:
The problem is that the index points to the last entry written upon close. Upon reopening the db, that first entry written overwrites that last entry. By incrementing the index upon open, the first entry written will write to the index after the last entry before save.

Reply:
I think we should add this as an inline comment, looking at this I too had no idea why it was being done.